chrome/thunderlink/content/thunderlink.js

```diff
@@ -44,43 +44,57 @@ var ThunderLinkChromeNS = {
     var appendToFileActive = prefService.getBoolPref("custom-tl-string-" + cstrnum + "-appendtofile-checkbox");
     console.log("ActivateCustomTlString: appendToFileActive: " + appendToFileActive + "\n");
 
+    var selectedMsgHeaders = gDBView.getSelectedMsgHdrs();
+    var selectionDelimiter = prefService.getCharPref("custom-tl-string-" + cstrnum + "-selection-delimiter");
+    var i = 0;
+
     // activate
     if (tagActive) {
-      this.TagEmail(prefService.getIntPref("custom-tl-string-" + cstrnum + "-tag"));
+      for (i = 0; i < selectedMsgHeaders.length; i++) {
+        this.TagEmail(prefService.getIntPref("custom-tl-string-" + cstrnum + "-tag"), selectedMsgHeaders[i]);
+      }
     }
+
     if (copyToClipboardActive) {
-      var procCustomTlStr = replaceVariables(customTlStr, gDBView.hdrForFirstSelectedMessage);
-      console.log("ActivateCustomTlString: procCustomTlStr resolved: " + procCustomTlStr + "\n");
+      var procCustomTlStr = "";
+      for (i = 0; i < selectedMsgHeaders.length; i++) {
+        procCustomTlStr += replaceVariables(customTlStr, selectedMsgHeaders[0]);
```

Comment:
Everything seems to work well except the '0' here should be an 'i' right? In my testing I get the same header each time because of this I believe

Reply:
You are right. Sorry for that :(